tests/test_hdate.py

```diff
@@ -80,6 +80,33 @@ def test_pesach_day_of_week(self, rand_hdate):
             assert rand_hdate.dow == info[2]
             assert rand_hdate.holiday_name == "pesach"
 
+    UPCOMING_SHABBATOT = [
+      ((2018, 11, 30), (2018, 12, 1), (5779, 3, 22)),
+      ((2018, 12, 1), (2018, 12, 1), (5779, 3, 23)),
+      ((2018, 12, 2), (2018, 12, 8), (5779, 3, 24)),
+      ((2018, 12, 3), (2018, 12, 8), (5779, 3, 25)),
+      ((2018, 12, 4), (2018, 12, 8), (5779, 3, 26)),
+      ((2018, 12, 5), (2018, 12, 8), (5779, 3, 27)),
+      ((2018, 12, 6), (2018, 12, 8), (5779, 3, 28)),
+      ((2018, 12, 7), (2018, 12, 8), (5779, 3, 29)),
+      ((2018, 12, 8), (2018, 12, 8), (5779, 3, 30)),
+      ((2018, 12, 9), (2018, 12, 15), (5779, 4, 1)),
+    ]
+    @pytest.mark.parametrize('current_date, shabbat_date, hebrew_date',
+                             UPCOMING_SHABBATOT)
+    def test_get_upcoming_shabbat_date(self, current_date, shabbat_date,
+                                       hebrew_date, rand_hdate):
+        rand_hdate.gdate = datetime.date(*current_date)
```

Comment:
No need for rand_hdate, use HDate directly. rand_hdate is to get a random Hebrew date